src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -540,7 +541,7 @@ public function mirror($originDir, $targetDir, \Traversable $iterator = null, $o
 
         // Iterate in destination folder to remove obsolete entries
         if ($this->exists($targetDir) && isset($options['delete']) && $options['delete']) {
-            $deleteIterator = $iterator;
+            $deleteIterator = 5 === \func_num_args() ? func_get_arg(4) : $iterator;
```

Comment:
```suggestion
            $deleteIterator = 4 < \func_num_args() ? func_get_arg(4) : $iterator;
```

We should catch this parameter even if the caller passes 42 arguments to this method.